File: plugins/itp-hooks/hooks/tests/pushover-budget-fixtures/go/pushover_with_escape_hatch.go
package main

import (
	"net/http"
	"net/url"
)

func sendPushoverBudgetOk(token, user, message string) error {
	payload := url.Values{
		"token":   []string{token},
		"user":    []string{user},
		"message": []string{message},
		"title":   []string{"Status: PUSHOVER-BUDGET-OK"},
	}

	_, err := http.PostForm("https://api.pushover.net/1/messages.json", payload)
	return err
}

File: plugins/itp-hooks/hooks/tests/pushover-budget-fixtures/typescript/pushover_with_escape_hatch.ts
async function sendProvenance(summary: string, details: string) {
  // PUSHOVER-BUDGET-OK: This message already uses structured format with URL codes
  const msg = {
    token: process.env.PUSHOVER_APP_TOKEN,
    user: process.env.PUSHOVER_USER_KEY,
    title: 'Deployment',
    message: `${summary}\nDEPLOY_ID=abc123\nBUILD_URL=https://ci.example.com/builds/456`,
    priority: 0,
  };

  const params = new URLSearchParams(msg as Record<string, string>);
  return fetch('https://api.pushover.net/1/messages.json', {
    method: 'POST',
    body: params,
  });
}
